frontend/resourceadm/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { ReactNode } from 'react';
+import classes from './Modal.module.css';
+import ReactModal from 'react-modal';
+
+/**
+ * Style the modal
+ */
+const modalStyles = {
+  content: {
+    width: '600px',
+    height: 'fit-content',
+    margin: 'auto',
+    padding: '32px',
+  },
+  overlay: {
+    backgroundColor: 'rgba(0, 0, 0, 0.7)',
+  },
+};
+
+interface Props {
+  isOpen: boolean;
+  title: string;
+  onClose?: () => void;
+  children: ReactNode;
+}
+
+/**
+ * Modal component implementing the react-modal.
```

Comment:
Hvis du gjør jsdoc av komponenten slik som vi har gjort i actionbar i stedet så vil man få forklaringen på selve propen på hover + at man får dokumentert selve komponenten og bruken av den

```suggestion
export interface ActionBarProps {
  /** Additional actions to be displayed on the right side of the ActionBar. */
  actions?: React.ReactNode;
  /** Additional text to be displayed on the right side of the header of the ActionBar. */
  additionalText?: React.ReactNode;
  /** The content to be displayed as expandable content inside the ActionBar. */
  children?: React.ReactNode;
  /** The color variant of the ActionBar. */
  color?: 'light' | 'neutral' | 'warning' | 'success' | 'danger';
  /** The click event handler for the ActionBar. */
  onClick?: ClickHandler;
  /** Specifies whether the ActionBar is open or closed. */
  open?: boolean;
  /** The subtitle to be displayed in the header of the ActionBar. */
  subtitle?: React.ReactNode;
  /** The title to be displayed in the header of the ActionBar. */
  title?: React.ReactNode;
}

/**
 * @component
 * @example
 * <ActionBar
 *    actions={
 *      <><button onClick={handleActionBarClick}>Action 1</button>
 *      <button onClick={handleActionBarClick}>Action 1</button></>
 *    }
 *    additionalText=<div>"Additional Text"</div>
 *    color="neutral"
 *    onClick={handleActionBarClick}
 *    open={openState}
 *    subtitle={<div>"Subtitle"</div>}
 *    title={<div>"Title"</div>}
 *   >
 *      <div>Content goes here</div>
 * </ActionBar>
 *
 * @property {React.ReactNode} [actions] - Additional actions to be displayed on the right side of the header of the ActionBar.
 * @property {React.ReactNode} [additionalText] - Additional text to be displayed in the header of the ActionBar.
 * @property {React.ReactNode} [children] - The content to be displayed as expandable content inside the ActionBar.
 * @property {'light' | 'neutral' | 'warning' | 'success' | 'danger'} [color='neutral'] - The color variant of the ActionBar.
 * @property {ClickHandler} [onClick] - The click event handler for the ActionBar.
 * @property {boolean} [open=false] - Specifies whether the ActionBar is open or closed.
 * @property {React.ReactNode} [subtitle] - The subtitle to be displayed in the header of the ActionBar.
 * @property {React.ReactNode} [title] - The title to be displayed in the header of the ActionBar.
 * @returns {React.ReactNode} The rendered ActionBar component.
 */
```

Reply:
Får opp forklaring på selve propen også ved å gjøre sånn her: 
![Screenshot 2023-06-21 at 16 57 06](https://github.com/Altinn/altinn-studio/assets/133344438/b8441768-fde7-4c43-9ce5-4c7df3902e7a)
Om det var sånn her du mente?

Har ikke lagt inn ```examples``` enda, men det er noe som vurderes til fremtiden :)

Reply:
Ja men hvis man legger på beskrivelse i propene og så må du ikke hovere over komponenten for å se prop-beskrivelser

Reply:
Aaah, sånn ja. Da skjønner jeg hva du mener. Det er absolutt som burde bli gjort her og 😄